src/main/java/hu/wurfel/refference/school/mark/MarkService.java

```diff
@@ -129,72 +139,72 @@ private void mWTeacher(MarkRequestForSearch markRequestForSearch) {
         }
     }
 
-    protected ArrayList<Mark> getMarksByDiary(@NotNull Diary diary) {
+    protected List<Mark> getMarksByDiary(@NotNull Diary diary) {
         return getMarksByDiaryid(diary.getId());
     }
 
-    protected ArrayList<Mark> getMarksByDiaries(@NotNull ArrayList<Diary> diaries) {
-        ArrayList<Mark> marks = new ArrayList<>();
+    protected List<Mark> getMarksByDiaries(@NotNull List<Diary> diaries) {
+        List<Mark> marks = new ArrayList<>();
         for (Diary diary : diaries) {
             marks.addAll(getMarksByDiary(diary));
         }
         return marks;
     }
 
-    protected ArrayList<Mark> getMarksByStudent(@NotNull Student student) {
-        ArrayList<Mark> marks = new ArrayList<>();
-        for (Diary diary : new ArrayList<Diary>(diaryCrudService.getDiariesByScid(student.getId()))) {
+    protected List<Mark> getMarksByStudent(@NotNull Student student) {
+        List<Mark> marks = new ArrayList<>();
+        for (Diary diary : new ArrayList<>(diaryCrudService.getDiariesByScid(student.getId()))) {
             marks.addAll(getMarksByDiary(diary));
         }
         return marks;
     }
 
-    protected ArrayList<Mark> getMarksByStudents(@NotNull ArrayList<Student> students) {
-        ArrayList<Mark> marks = new ArrayList<>();
+    protected List<Mark> getMarksByStudents(@NotNull List<Student> students) {
+        List<Mark> marks = new ArrayList<>();
         for (Student student : students) {
             marks.addAll(getMarksByStudent(student));
         }
         return marks;
     }
```

Comment:
Értem, hogy miért így csináltad. Annyi hint, hogy ezt meg lehet csinálni a repo által generált queryvel, így nem kell iterálni vagy csak egyszer, ezt most nem tudom hirtelen. Ez alapvetően a db dolga, nem a service-é. sokkal gyorsabb is úgy.